tests/PrefixCacheBootstrapperTest.php

```diff
@@ -0,0 +1,345 @@
+
+    $tenant1 = Tenant::create();
+    $tenant2 = Tenant::create();
+    tenancy()->initialize($tenant1);
+
+    // The store isn't prefixed, so the cache isn't separated
+    expect(cache()->store($cacheStore)->get('key'))->toBe('central-value');
+    $this->app->make(SpecificCacheStoreService::class)->handle();
+    expect(cache()->store($cacheStore)->get('key'))->toBe($tenant1->getTenantKey());
+
+    tenancy()->initialize($tenant2);
+
+    expect(cache()->store($cacheStore)->get('key'))->toBe($tenant1->getTenantKey());
+    $this->app->make(SpecificCacheStoreService::class)->handle();
+    expect(cache()->store($cacheStore)->get('key'))->toBe($tenant2->getTenantKey());
+
+    tenancy()->end();
+    expect(cache()->store($cacheStore)->get('key'))->toBe($tenant2->getTenantKey());
+});
+
+test('stores specified in tenantCacheStores get prefixed', function() {
+    // Make the currently used store ('redis') the only store in $tenantCacheStores
+    PrefixCacheTenancyBootstrapper::$tenantCacheStores = ['redis'];
+
+    $this->app->singleton(CacheService::class);
+
+    $this->app->make(CacheService::class)->handle();
+    expect(cache('key'))->toBe($centralValue = 'central-value');
+
+    $tenant1 = Tenant::create();
+    $tenant2 = Tenant::create();
+
+    tenancy()->initialize($tenant1);
+
+    expect(cache('key'))->toBeNull();
+    $this->app->make(CacheService::class)->handle();
+    expect(cache('key'))->toBe($tenant1->getTenantKey());
+
+    tenancy()->initialize($tenant2);
+
+    expect(cache('key'))->toBeNull();
+    $this->app->make(CacheService::class)->handle();
+    expect(cache('key'))->toBe($tenant2->getTenantKey());
+
+    tenancy()->end();
+    expect(cache('key'))->toBe($centralValue);
+});
+
+test('stores not specified in tenantCacheStores do not get prefixed', function() {
+    config(['cache.stores.redis2' => config('cache.stores.redis')]);
+    config(['cache.default' => 'redis2']);
+    // Make 'redis' the only store in $tenantCacheStores so that the current store ('redis2') doesn't get prefixed
+    PrefixCacheTenancyBootstrapper::$tenantCacheStores = ['redis'];
+
+    $this->app->singleton(CacheService::class);
+
+    $this->app->make(CacheService::class)->handle();
+    expect(cache('key'))->toBe('central-value');
+
+    $tenant1 = Tenant::create();
+    $tenant2 = Tenant::create();
+    tenancy()->initialize($tenant1);
+
+    // The cache isn't prefixed, so it isn't separated
+    expect(cache('key'))->toBe('central-value');
+    $this->app->make(CacheService::class)->handle();
+    expect(cache('key'))->toBe($tenant1->getTenantKey());
+
+    tenancy()->initialize($tenant2);
+
+    expect(cache('key'))->toBe($tenant1->getTenantKey());
+    $this->app->make(CacheService::class)->handle();
+    expect(cache('key'))->toBe($tenant2->getTenantKey());
+
+    tenancy()->end();
+    expect(cache('key'))->toBe($tenant2->getTenantKey());
+});
+
+test('non default stores get prefixed too', function () {
```

Comment:
What exactly does this test check? That even stores that aren't default, **but are specified in `tenantCacheStores`**, will get prefixed? If so, the name should be updated since right now it sounds like they get prefixed automatically.

Either something like "non default stores can be prefixed as well" or "non default stores get prefixed too when listed in tenantCacheStores"

Reply:
Yeah, you got it right. I agree with the name change – I'd go with something like the latter suggestion